lib/ansible/module_utils/network/skydive/api.py

```diff
@@ -161,3 +280,66 @@ def run(self, ib_spec):
                 result['changed'] = True
 
         return result
+
+
+class skydive_node(skydive_wsclient, skydive_restclient):
+    """ Implements Skydive Node modules """
+
+    def __init__(self, module):
+        self.module = module
+        provider = module.params['provider']
+        super(skydive_node, self).__init__(self.module, **provider)
+
+    def run(self):
+        try:
+            lookup_query = SKYDIVE_GREMLIN_QUERY + "('Name', '{0}')".format(self.module.params['name'])
```

Comment:
No you cannot consider names as unique. As said in a previous commit, you will probably have a lot of "eth0" named nodes.

Reply:
Added `Type` field in the check along with the `Name` param.